### DISCUSSION TITLE
Any actual development ...

### DISCUSSION THREAD

**Discussion by MartinNK** [RaBa64/OpenDTU#15] [General]
Hi Ralf,

I am very interested in further development, maybe I will be able to help with testing and a little bit in development. I am Running 2 openDTU Database systems on ESP32 with different Hoymile HM-xxx.

Happy about your project and hope you will go on.

What about getting rid off google charts? Any update on getting a pull of actual openDTU now?

Martin

**Comment by BigT71** [comment]
Me too.

can you build a new release, only to add the new code from the main branch?

**Comment by kinichahau2023** [comment]
Yes please !!! 
I am also very happy with your project and looking forward to the next update.

What about restoring the database (not only save ad it is working now) after update? Or remains the database "as is" during the update untouched?

Regards, Andreas

**Reply by RaBa64** [reply]
Ja, die Daten bleiben beim Update erhalten.

**Comment by MartinNK** [comment]
Anybody her? @RaBa64 are you here?

**Reply by kinichahau2023** [reply]
we are still waiting for Ralf's next development... let's hope that he still continues this beside his other projects as well.

**Reply by MartinNK** [reply]
Nothing new since 2 Months ... feeling sad.

**Reply by RaBa64** [reply]
Sorry guys, I am a little bit busy at the moment. I am building a large 10kWp PV on my roof.
For that I had to disconnect my Hoymiles inverter and cannot test anything at the moment.
I will try to do at least an update to the current master of OpenDTU.
I hope that helps.

**Reply by kinichahau2023** [reply]
Yes please... that would be excellent.
Thx in advance.

**Reply by MartinNK** [reply]
@RaBa64 vielen Dank fürs Update. Viel Erfolg für deine PV Anlage, bin gespannt was Du drüber erzählen wirst. Wäre schön mal etwas über deine Konfiguration zu erfahren. 10kWp ist eine ja Hausnummer ... wohl aber ohne Hoymiles und openDTU ab nun?
Kleine Frage zur neuen Version ... nur eine Pull der Original openDTU oder sind dort kleinere Änderungen drin? Bin leider noch aus der alten Schule und tue mich schwer mit dem ganzen C++, JSON, JS u.a.

**Comment by BigT71** [comment]
@RaBa64 vielen Dank!

**Comment by MartinNK** [comment]
@RaBa64 Danke für die Aktualisierung für 2024. Wie schierig ist es eigentlich die Grafiken anzupassen damit diese auch auf einem Mobilengerät richtig anzuzeigen bzw. bei schmalem Browserfenster auf die Breite des Fensters anzupassen?
Was macht die Solaranlage? Bin mal gespannt über einen kleinen Bericht Hardware und Software ... alles komerzielle oder auch offene Systeme verwendet? Grüß vom Bodensee.

**Reply by XpertXXL** [reply]
vielen Dank, dass wir nun auch die Daten vom aktuellen Jahr angezeigt bekommen :-)

**Reply by RaBa64** [reply]
Leider habe ich keine Möglichkeit gefunden, dass sich der Google-Charts Kalender automatisch so anpasst, dass er sowohl auf einem kleinen als auch großen Bildschirm gut aussieht.
Ich versuche gerade diesen Kalender zu integrieren:
https://razorness.github.io/vue3-calendar-heatmap/

**Reply by XpertXXL** [reply]
Hallo RaBa64, das ging ja mal sauschnell, sieht übersichtlich aus und ist responsiv :-)
Zeil erreicht!!!
vielen Dank dafür :-)

**Reply by MartinNK** [reply]
Danke für den tollen neuen Kalender. Leider hab ich bei den letzten Aktionen wohl die Daten auf meiner DTU geschrottet und Database nun wieder auf Null. Schön wäre diese wieder einspielen zu können. Bin auch mal gespannt wie die Visualisierung später aussieht ohne Google Chart.

**Comment by RaBa64** [comment]
> @RaBa64 Danke für die Aktualisierung für 2024. Wie schierig ist es eigentlich die Grafiken anzupassen damit diese auch auf einem Mobilengerät richtig anzuzeigen bzw. bei schmalem Browserfenster auf die Breite des Fensters anzupassen? Was macht die Solaranlage? Bin mal gespannt über einen kleinen Bericht Hardware und Software ... alles komerzielle oder auch offene Systeme verwendet? Grüß vom Bodensee.



> @RaBa64 Danke für die Aktualisierung für 2024. Wie schierig ist es eigentlich die Grafiken anzupassen damit diese auch auf einem Mobilengerät richtig anzuzeigen bzw. bei schmalem Browserfenster auf die Breite des Fensters anzupassen? Was macht die Solaranlage? Bin mal gespannt über einen kleinen Bericht Hardware und Software ... alles komerzielle oder auch offene Systeme verwendet? Grüß vom Bodensee.

**Comment by RaBa64** [comment]
Mit dem neuen Calendar View sollte jetzt alles auch auf dem Smartphone funktionieren.

Meine neue PV-Anlage ist fertig und läuft soweit sehr gut.
10,56kWp mit Trina n-Type Modulen (à 440Wp) / 16kWh Batterie / Sungrow Wechselrichter.
Jetzt muss ich aber erst einmal auf den Zählerwechsel warten, bis ich sie in Betrieb nehmen darf :-(

**Reply by MartinNK** [reply]
Hi Ralf,
Mobil scheint jetzt auch zu funktionieren, die Agenda skaliert aber nicht richtig. Browser Firefox Mobil oder Desktop.
Hier nochmal zur neuen Ansicht ...

![Screenshot_20240109_183808](https://github.com/RaBa64/OpenDTU-Database/assets/3524916/2ca9afd9-675d-4818-8140-068e497e61e3)

Hier fehlt mir wie gesagt die Einteilung / Unterteilung des Kalender ... links unten Less / More da fehlt mir Min / Max in Zahlen ... Monatskürzel zentriert über dem Monat wäre schöner ... Wochentage 1. Buchstabe reicht ... Fliesender Kalender wie ich das sehe und da fehlt irgendwie schon Jahreszahl vorne / hinten (klar bei Januar bis Dezember ist es das aktuelle Jahr. Tooltip hab ich gesehen.

![Screenshot_20240109_183746](https://github.com/RaBa64/OpenDTU-Database/assets/3524916/372a9f0c-7a37-4534-a084-85200df4f271)

Finde den alten von der Übersichtlichkeit besser ... Abgrenzung und Orientierung ... mal abgesehen von den alten Problemen die er gemacht hat ;-)

**Comment by MartinNK** [comment]
Hi @RaBa64 Ralf,

wie sieht es aus mit einer neuen Version? Im Moment läuft es auc mit Google charts ganz gut ... meine System läuft nun bald 1 Jahr mit einer Database ... angezeigt wird ja max. 1 Jahr aber in die Database passen nach deinen Berechnungen auch gerne >2 Jahre.

Klar könnte mir eine Database editieren und dann eine bis 31.12.2023 und eine andere ab dem 1.1.2024 machen.

Was passiert im Moment wenn ich aber eine komplette Database behalten will ... klar es wir einmal ab der akt. Woche nur 1 Jahr rückwärte im Kalender angezeigt ... wäre schön wenn wir da eine Möglichkeit fürs scrollen schaffen könnten.

Bist Du noch ein bissle weiter gekommen wegen der Google charts?

Grüße
Martin

**Reply by kinichahau2023** [reply]
Hallo Ralf,
Kannst Du mal wieder ein Update generieren? 
Es gab doch zahlreiche Neuerungen im Original ... 
Gruß Andreas
